js/controller/messagecontroller.js

```diff
@@ -120,13 +120,14 @@ define(function(require) {
 	 * @param {number} attachmentId
 	 * @returns {Promise}
 	 */
-	function saveAttachmentToFiles(account, folder, messageId, attachmentId) {
+	function saveAttachmentToFiles(account, folder, messageId, attachmentId, callback) {
```

Comment:
@haivp3010 don't know whether you're familiar with the concept of promises, but this function returns one. This means we get an object that we can register callback on, which are executed as soon as the asynchronous task has finished. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises for more info 😉

Reply:
I do know about promises, but in this case, I just haven't found out if that close event of the file chooser calls the reject method of the promise.

Reply:
I don't know that either as I haven't checked. But from the diff here on GitHub I can tell that you're invoking the callback as soon as the user selected a directory. However, the loading feedback should be there until the task of actually storing the attachments has been finished.

Reply:
Wait. Ignore my comment, now I see what you did there 🙈 The callback was added for *adding* the feeback, not for removing.

Reply:
Yeah. That's what I meant: no loading until user has chosen a directory, and loading disappears after upload done. I have checked that on my localhost and it went fine. It's just that I couldn't find a way to achieve this with promise, with the existing code. I had thought of catching the close event on the file chooser and call reject, but that would require making changes to core js files of nextcloud server as well, so I had to resort to callback like that.